gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/adapter/ApiAdapter.java

```diff
@@ -84,6 +84,7 @@ public interface ApiAdapter {
     @Mapping(target = "tags", source = "api.tags")
     @Mapping(target = "apiVersion", source = "api.version")
     @Mapping(target = "lifecycleState", source = "api.apiLifecycleState")
+    @Mapping(target = "listeners", ignore = true)
```

Comment:
I think your test failed because of this. no ? 

What about specifying listeners like this `@Mapping(target = "listeners", source = "apiDefinitionV4.listeners")` ?